newIDE/app/src/Profile/Subscription/SubscriptionDialog.js

```diff
@@ -121,6 +131,25 @@ const cancelAndChangeWithValidRedeemedCodeConfirmationTexts = {
   maxWidth: 'sm',
 };
 
+const getSubscriptionPricingSystemPeriod = (
+  subscription: ?Subscription,
+  subscriptionPlansWithPricingSystems: ?(SubscriptionPlanWithPricingSystems[])
+): null | 'year' | 'month' => {
+  if (!subscription || !subscriptionPlansWithPricingSystems) return null;
+  for (let subscriptionPlanWithPricingSystems of subscriptionPlansWithPricingSystems) {
+    for (let pricingSystem of subscriptionPlanWithPricingSystems.pricingSystems) {
+      if (pricingSystem.id === subscription.pricingSystemId) {
+        if (pricingSystem.period === 'week') {
+          // TODO: Add support for weekly subscriptions when needed.
```

Comment:
do we really want that? 😅

Reply:
Maybe. Or I could remove `'week'` from the typing. But I'd rather have it and not make the app bug in case we want to add it one day. This way, backend changes won't affect the app